cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -263,6 +326,75 @@ def maybe_install_ua_tools(cloud):
         raise
 
 
+def _should_auto_attach(ua_section: dict) -> bool:
+    disable_auto_attach = bool(
+        ua_section.get("features", {}).get("disable_auto_attach", False)
+    )
+    if disable_auto_attach:
+        return False
+
+    try:
+        from uaclient.api.exceptions import UserFacingError
+        from uaclient.api.u.pro.attach.auto.should_auto_attach.v1 import (
+            should_auto_attach,
+        )
+    except ImportError as ex:
+        LOG.debug("Unable to import `uaclient`: %s", ex)
+        LOG.warning(ERROR_MSG_SHOULD_AUTO_ATTACH)
+        return False
+    try:
+        result = should_auto_attach()
+    except UserFacingError as ex:
+        LOG.debug("Error during `should_auto_attach`: %s", ex)
+        LOG.warning(ERROR_MSG_SHOULD_AUTO_ATTACH)
+        return False
+    return result.should_auto_attach
+
+
+def _attach(ua_section: dict):
+    token = ua_section.get("token")
+    if not token:
+        msg = "`ubuntu-advantage.token` required in non-Pro Ubuntu instances."
+        LOG.error(msg)
+        raise RuntimeError(msg)
+    enable_beta = ua_section.get("enable_beta")
+    if enable_beta:
+        LOG.debug(
+            "Ignoring `ubuntu-advantage.enable_beta` services in UA attach:"
+            " %s",
+            ", ".join(enable_beta),
+        )
+    configure_ua(
+        token=token,
+        enable=ua_section.get("enable"),
+        config=ua_section.get("config"),
```

Comment:
@aciba90 I just realized one big gap in this implementation is that we are only invoking `ua config *http_proxy*` values on non-PRO instances. I think we actually need this feature to override proxy config for both PRO and non-PRO instances, so this may involve a bit of a refactor to ensure both PRO instances and can still manipulate proxy values prior to auto-attach of plain attach.